bin/fin

```diff
@@ -119,10 +119,33 @@ is_katacoda ()
 	[[ "$KATACODA" != "" ]]
 }
 
-is_github_actions () {
+is_github_actions ()
+{
 	[[ "$GITHUB_RUN_ID" != "" ]]
 }
 
+# Returns x86_64 / aarch64 (aka "old school")
+host_arch () {
+	case "$(uname -m)" in
+		x86_64) echo "x86_64" ;;
+		amd64) echo "x86_64" ;;
+		aarch64) echo "aarch64" ;;
+		arm64) echo "aarch64" ;;
+		* ) false;;
+	esac
+}
```

Comment:
```suggestion
host_arch () {
    case "$(uname -m)" in
        aarch64|arm64) echo "aarch64";;
        *) echo "x86_64";;
    esac
}
```

Reply:
If the host is neither x86_64, nor aarch64, then defaulting to x86_64 would not help as nothing else will work anyway. I'd rather have this function return `false` if host arch is not expected and add a check at the beginning of runtime:

```bash
if ! host_arch; then 
    echo "Host architecture not supported."
    exit 1
fi
```

Reply:
@achekulaev I added the host architecture check and also made it so that `host_arch`/`host_arch_new` always return a string value (returning `uname -m` in the catch-all case).